### ISSUE TITLE
Remove force trainable condition in AveragedOptimizerWrapper

### ISSUE THREAD

**Issue by fsx950223** [tensorflow/addons#2245]
**Describe the feature and the current behavior/state.**
Could remove https://github.com/tensorflow/addons/blob/26eb3d2f791a747bd62a8c5227476e04ed9e8e49/tensorflow_addons/optimizers/average_wrapper.py#L126
trainable condition?
Some layers variables are not trainable(such as BatchNormalization moving_variance and moving_mean), I want to add these variables to moving average variables too.
**Relevant information**
- Are you willing to contribute it (yes/no):
- Are you willing to maintain it going forward? (yes/no):
- Is there a relevant academic paper? (if so, where):
- Is there already an implementation in another framework? (if so, where):
- Was it part of tf.contrib? (if so, where):

**Which API type would this fall under (layer, metric, optimizer, etc.)**
optimizer
**Who will benefit with this feature?**

**Any other info.**

**Comment by WindQAQ** [comment]
Yep, I think the optimizer should be trainable ignostic, and it should process all vars in `var_list` passed into it.

**Comment by fsx950223** [comment]
Current api strongly bound with apply_gradients which is not  flexible. Is it possible to break the rule and update ema via assign_average_vars

**Comment by fsx950223** [comment]
It's not necessary.
